ipm_service/test/test_ipm_service.py

```diff
@@ -0,0 +1,267 @@
+    camera_info = CameraInfo(
+        header=Header(frame_id="camera_optical_frame"),
+        width=2048,
+        height=1536,
+        binning_x=4,
+        binning_y=4,
+        k=[1338.64532, 0., 1026.12387, 0., 1337.89746, 748.42213, 0., 0., 1.])
+    camera_info_pub.publish(camera_info)
+    rclpy.spin_once(ipm_service_node, timeout_sec=0.1)
+
+    point = Point2DStamped(
+        header=Header(frame_id="camera_optical_frame"),
+        point=Point2D(x=100.0, y=100.0))
+
+    # XY-plane at z = 1.0
+    # Create Plane in the same frame as our camera with 1m distance facing the camera
+    plane = PlaneStamped()
+    plane.header.frame_id = "camera_optical_frame"
+    plane.plane.coef[2] = 1.0  # Normal in z direction
+    plane.plane.coef[3] = -1.0  # 1 meter distance
+
+    client = test_node.create_client(MapPoint, 'map_point')
+    req = MapPoint.Request(point=point, plane=plane)
+    future = client.call_async(req)
+    rclpy.spin_once(ipm_service_node, timeout_sec=0.1)
+
+    rclpy.spin_once(test_node, timeout_sec=0.1)
+
+    assert future.result() is not None
+    assert future.result().result == MapPoint.Response.RESULT_SUCCESS
+
+    ipm = IPM(Buffer(), camera_info)
+    expected_point = ipm.map_point(plane, point)
+    assert future.result().point == expected_point
+
+    rclpy.shutdown()
+
+
+def test_map_point_cloud_no_camera_info():
+
+    rclpy.init()
+    ipm_service_node = IPMService()
+    test_node = rclpy.node.Node('test')
+
+    client = test_node.create_client(MapPointCloud2, 'map_pointcloud2')
+    future = client.call_async(MapPointCloud2.Request())
+    rclpy.spin_once(ipm_service_node, timeout_sec=0.1)
+
+    rclpy.spin_once(test_node, timeout_sec=0.1)
+
+    assert future.result() is not None
+    assert future.result().result == MapPointCloud2.Response.RESULT_NO_CAMERA_INFO
+
+    rclpy.shutdown()
+
+
+def test_map_point_cloud_invalid_plane():
+
+    rclpy.init()
+    ipm_service_node = IPMService()
+    test_node = rclpy.node.Node('test')
+
+    camera_info_pub = test_node.create_publisher(CameraInfo, 'camera_info', 10)
+    camera_info = CameraInfo()
+    camera_info_pub.publish(camera_info)
+    rclpy.spin_once(ipm_service_node, timeout_sec=0.1)
+
+    point_cloud = create_cloud(
+        header=Header(),
+        fields=[
+            PointField(name='x', offset=0, datatype=PointField.FLOAT32, count=1),
+            PointField(name='y', offset=4, datatype=PointField.FLOAT32, count=1)],
+        points=[])
+
+    client = test_node.create_client(MapPointCloud2, 'map_pointcloud2')
+    # Request with the default plane a=b=c=0 should be an invalid plane
+    future = client.call_async(MapPointCloud2.Request(points=point_cloud))
+    rclpy.spin_once(ipm_service_node, timeout_sec=0.1)
+
+    rclpy.spin_once(test_node, timeout_sec=0.1)
+
+    assert future.result() is not None
+    assert future.result().result == MapPointCloud2.Response.RESULT_INVALID_PLANE
+
+    rclpy.shutdown()
+
```

Comment:
You seemed to exclude the ``test_project_point_cloud_no_intersection_error()`` here, was that intentional?

Reply:
Yeah, as point clouds include many points a single no intersection error is not very useful. Especially if most of them are intersecting. Said exception is only thrown for the single point mapping. The current solution sets them to nan if they do not intersect.

Reply:
Good point, we should then get rid of ``uint8 RESULT_ON_INTERSECTION=3`` from ProjectPointCloud2.srv, either in this PR or another one.

Reply:
I think we should do this in a different one

Reply:
Moving this to an issue: #31